shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteCardData.kt

```diff
@@ -264,33 +266,34 @@ data class RouteCardData(
             now: Instant,
             pinnedRoutes: Set<String>,
             context: Context
-        ): List<RouteCardData>? {
+        ): List<RouteCardData>? =
+            withContext(Dispatchers.Default) {
 
-            // if predictions or alerts are still loading, this is the loading state
-            if (predictions == null || alerts == null) return null
+                // if predictions or alerts are still loading, this is the loading state
+                if (predictions == null || alerts == null) return@withContext null
 
-            // if global data was still loading, there'd be no nearby data, and null handling is
-            // annoying
-            if (globalData == null) return null
+                // if global data was still loading, there'd be no nearby data, and null handling is
+                // annoying
+                if (globalData == null) return@withContext null
 
-            val hideNonTypicalPatternsBeyondNext: Duration? =
-                when (context) {
-                    Context.NearbyTransit -> 120.minutes
-                    Context.StopDetailsFiltered -> 120.minutes
-                    Context.StopDetailsUnfiltered -> null
-                }
+                val hideNonTypicalPatternsBeyondNext: Duration? =
+                    when (context) {
+                        Context.NearbyTransit -> 120.minutes
+                        Context.StopDetailsUnfiltered -> 120.minutes
+                        Context.StopDetailsFiltered -> null
```

Comment:
There's an acceptance criterion that says that unfiltered stop details should have the same rules as nearby transit, which doesn't match what was previously implemented here.